src/Servers/Kestrel/shared/TransportConnection.Generated.cs

```diff
@@ -137,23 +138,23 @@ private void ExtraFeatureSet(Type key, object? value)
 
                 if (key == typeof(IConnectionIdFeature))
                 {
-                    _currentIConnectionIdFeature = value;
+                    _currentIConnectionIdFeature = (IConnectionIdFeature?)value;
```

Comment:
@benaadams shouldn't the unsafe cast be here too?

Reply:
Its the `this[Type key] = object` version; so can't guarantee `value` is the correct type so Unsafe cast here would be bad

Reply:
It occurs to me that we had this bug today where you could basically store anything in here and it would explode on get